groups/models.py

```diff
@@ -13,6 +13,13 @@ class CommentManager(models.Manager):
     def for_discussion_pk(self, pk):
         return self.get_queryset().filter(discussion_id=pk)
 
+    def with_user_may_delete(self, user):
+        queryset = self.get_queryset()
+        for comment in queryset:
+            comment.user_may_delete = comment.may_be_deleted(user)
+
+        return queryset
```

Comment:
Yes, but you can `filter` (etc) a `queryset` again which does a new sql query. I'm worried that this would lose the `user_may_delete` attribute.